samples/stream_download_percentage/src/sample/stream_download_percentage.cljd

```diff
@@ -0,0 +1,164 @@
+(ns sample.stream-download-percentage
+  "Example for using drawer with named routes navigation."
+  (:require ["package:flutter/material.dart" :as m]
+            ["package:path_provider/path_provider.dart" :as path-provider]
+            ["dart:core" :as dc]
+            ["dart:io" :as dio]
+            ["package:http/http.dart" :as http]
+            ["package:validators/validators.dart" :as validators]
+            [cljd.flutter :as f]
+            [clojure.string :as str]))
+
+(defn calculate-percentage
+  [{::keys [total
+            completion]}]
+  (if (= 0 total)
+    0
+    (-> (- total
+           completion)
+        (/ total)
+        (* 100)
+        (- 100)
+        (* -1))))
+
+
+(defn delete-local-file!
+  [file-name]
+  (let [old-file (-> file-name
+                     dio/File)]
+    (when (await (.exists old-file))
+      (-> old-file
+          .delete))))
+
+
+(defn parse-uri
+  [tc]
+  (let [form-text (-> tc
+                      .-text)]
+    (when (validators/isURL form-text)
+      (-> form-text
+          Uri/parse))))
+
+
+(defn start-button-handler
+  [{::keys [file-name
+            request-uri
+            *total
+            *completion
+            file-image-atom]}]
+  (delete-local-file! file-name)
+  (let [streamed-response (some->> request-uri
+                                   (http/Request "get")
+                                   .send
+                                   await)
+        local-file-open (dio/File file-name)
+        opened-file (.openWrite local-file-open)
+        _total-bytes-add (swap! *total
+                                #(+ % (.-contentLength streamed-response)))]
+    (some-> streamed-response
+            .-stream
+            (.listen (fn [event]
```

Comment:
Rather than `.listen .onDone` you should consider using `.forEach` or `.fold` and `await` on the resulting future for termination.